### PR TITLE
eos-paygd: Don't enforce PAYG if Secure Boot is off

### PR DESCRIPTION
**Pull request by mwleeds**
If the current boot occurred with Secure Boot disabled, it doesn't make
sense to enforce PAYG because (a) it's easily circumventable by
replacing the initramfs and (b) Secure Boot has likely been disabled
because the machine is in one of two states:
1. It has been fully paid off and unlocked (though note that it's also
possible to have Secure Boot enabled on an unlocked machine, in which
case EOSPAYG_active will not be set).
2. It is being debugged by a developer who has used PKpriv to disable
Secure Boot.

This commit also adds a command line flag --skip-sb-check, so a
developer can have eos-paygd in enforcing mode even with Secure Boot
off (though adding the flag would require editing the initramfs or using
a specially created debug image).

https://phabricator.endlessm.com/T27735

### CODE REVIEW

File: eos-paygd/main.c [hunk 1]
```diff
@@ -231,11 +233,36 @@ test_and_update_securitylevel (char *procname)
...
+static gboolean
+secure_boot_enabled (void)
+{
+  uint8_t *secboot = NULL;
+  size_t data_size = 0;
+  uint32_t attributes;
+  int ret;
+
+  ret = efi_get_variable (EFI_GLOBAL_VARIABLE_GUID, "SecureBoot", &secboot, &data_size, &attributes);
+  if (ret < 0 || !secboot || data_size != 1)
```
Comment (ManMower) [inline eos-paygd/main.c]: !secboot just means secure boot is off here, doesn't it? It should be left to the other conditional

File: eos-paygd/main.c [hunk 2]
```diff
@@ -231,11 +233,36 @@ test_and_update_securitylevel (char *procname)
...
+  return TRUE;
+}
+
 static gboolean print_level = FALSE;
+static gboolean skip_sb_check = FALSE;
 
 static GOptionEntry opts[] =
 {
   { "seclevel", 's', 0, G_OPTION_ARG_NONE, &print_level, "Print security level and exit", NULL },
+  { "skip-sb-check", 0, 0, G_OPTION_ARG_NONE, &print_level, "Enforce PAYG even if Secure Boot is off", NULL },
```
Comment (ManMower) [inline eos-paygd/main.c]: s/print_level/skip_sb_check/

File: eos-paygd/main.c (hunk 1 above)
Comment (mwleeds) [reply eos-paygd/main.c]: No, looking at the implementation in libefivar, I think it's just never the case that `ret == 0 && secboot == NULL`. And experimentally my machine has SB off and on it `secboot != NULL && *secboot == 0`. Side note: I changed the conditional below from `secboot == 0` to `*secboot == 0`.

File: eos-paygd/main.c (hunk 2 above)
Comment (mwleeds) [reply eos-paygd/main.c]: Oops

File: eos-paygd/main.c (hunk 1 above)
Comment (ManMower) [reply eos-paygd/main.c]: Oops, you're right, the other check was the bug, and you've fixed it.

**Pull request merged**